core/src/helpers.ts

```diff
@@ -28,6 +28,33 @@ const upperFirst = (value: string): string =>
 const lowerFirst = (value: string): string =>
   value.charAt(0).toLowerCase() + value.slice(1);
 
+// eslint-disable-next-line  @typescript-eslint/no-explicit-any
+function deleteKeyFromObject(inputObject: any, keyToDelete: string) {
+  for (let [currentObjectKey, currentObjectValue] of Object.entries(
+    inputObject
+  )) {
+    if (currentObjectKey === keyToDelete) {
+      delete inputObject[keyToDelete];
+    } else if (Array.isArray(currentObjectValue)) {
```

Comment:
otherwise, if the key's value is an array, send it to the fn that handles arrays